tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -438,12 +442,12 @@ private[optimizer] abstract class OptimizerCore(
           case _                     => DoWhile(newBody, newCond, None)
         }
 
-      case Try(block, errVar, EmptyTree, finalizer) =>
+      case TryFinally(block, finalizer) =>
         val newBlock = transform(block, isStat)
         val newFinalizer = transformStat(finalizer)
-        Try(newBlock, errVar, EmptyTree, newFinalizer)(newBlock.tpe)
+        TryFinally(newBlock, newFinalizer)(newBlock.tpe)
```

Comment:
Consider moving the `TryFinally` case after the `TryCatch` case, so that we maintain the same order everywhere.